x-pack/plugins/observability_solution/asset_manager/server/routes/enablement/enable.ts

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  readEntityDiscoveryAPIKey,
+  saveEntityDiscoveryAPIKey,
+  deleteEntityDiscoveryAPIKey,
+  checkIfAPIKeysAreEnabled,
+  canEnableEntityDiscovery,
+  generateEntityDiscoveryAPIKey,
+  checkIfEntityDiscoveryAPIKeyIsValid,
+} from '../../lib/auth';
+import { RequestHandlerContext } from '@kbn/core/server';
+import { SetupRouteOptions } from '../types';
+import { ENTITY_INTERNAL_API_PREFIX } from '../../../common/constants_entities';
+import { EntityDiscoveryApiKeyType } from '../../saved_objects';
+
+export function enableEntityDiscoveryRoute<T extends RequestHandlerContext>({
+  router,
+  server,
+}: SetupRouteOptions<T>) {
+  router.put(
+    {
+      path: `${ENTITY_INTERNAL_API_PREFIX}/enablement`,
+      validate: false,
+    },
+    async (context, req, res) => {
+      try {
+        server.logger.debug("checking if API key service is enabled");
+
+        const apiKeysEnabled = await checkIfAPIKeysAreEnabled(server);
+        if (!apiKeysEnabled) {
+          return res.ok({
+            body: {
+              success: false,
+              reason: "API key service is not enabled; try configuring `xpack.security.authc.api_key.enabled` in your elasticsearch config",
+            }
+          });
+        };
+
+        server.logger.debug("checking if current Kibana user has permission to enable entity discovery");
+
+        const esClient = (await context.core).elasticsearch.client;
```

Comment:
nit we could pass `elasticsearch.client.asCurrentUser` to downstream functions so we don't use internal user accidentally